### PR TITLE
refactor(components): [backtop]

### PR DESCRIPTION
**Pull request by jw-foss**
* Extract logic code from component itself.
* Reorganize code for better readability.

This is part of the upcoming headless element-plus, by extracting the common logic out of the UI itself it will enable our users to use the logic and do their own customization on the UI.

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Review by holazz — APPROVED** [review]

File: packages/components/backtop/src/use-backtop.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        throwError(componentName, `target does not exist: ${props.target}`)
+      }
+      container.value = el.value
+    }
+  })
+
+  return {
+    visible,
+    handleClick,
+    handleScrollThrottled,
```
Comment (ryuhangyeong) [inline packages/components/backtop/src/use-backtop.ts:63]: ## Question

Is there any reason to export `handleScrollThrottled`?

Comment (jw-foss) [reply packages/components/backtop/src/use-backtop.ts:63]: @ryuhangyeong nice catch, I think we do not need to return this. 👍

**Review by ryuhangyeong — APPROVED** [review]

File: packages/components/backtop/src/use-backtop.ts (hunk 1 above)
Comment (jw-foss) [reply packages/components/backtop/src/use-backtop.ts]: @ryuhangyeong Updated! :bow: PTAL.

Comment (ryuhangyeong) [reply packages/components/backtop/src/use-backtop.ts]: @jw-foss ✅ 🤗

**Pull request merged**